### PR TITLE
enforcing MicroInterpreter buffer api specifications in Kernal Runner

### PR DESCRIPTION
**Pull request by turbotoribio**
MicroInterpreter specify what kind of buffer api is available but this is only tested in integration test. Changing Kernal Runner enforce same buffer api policy at kernel test level instead of only checking for these buffer api specifications in integration test.

Corresponding Internal Changes - http://cl/453570499

bug=[b/232283593](https://b.corp.google.com/issues/232283593)

### CODE REVIEW

File: tensorflow/lite/micro/kernels/kernel_runner.cc [hunk 1]
```diff
@@ -92,4 +98,4 @@ TfLiteStatus KernelRunner::Invoke() {
 }
 
 }  // namespace micro
-}  // namespace tflite
+}  // namespace tflite
```
Comment (tingyan19) [inline tensorflow/lite/micro/kernels/kernel_runner.cc:101]: Is this change intentional/necessary?

Comment (turbotoribio) [reply tensorflow/lite/micro/kernels/kernel_runner.cc:101]: wait I'm confused which change line 101 ?

**Review by tingyan19 — COMMENTED** [review]
There are some test failures. I think it might be because the test does not conform to the new restriction you added? Could you take a look at the CI failures and correct them?

**Comment by turbotoribio** [comment]
> There are some test failures. I think it might be because the test does not conform to the new restriction you added? Could you take a look at the CI failures and correct them?

I believe I fixed it !

**Comment by turbotoribio** [comment]
Accidentally closed PR...

**Review by advaitjain — APPROVED** [review]
